src/webp_converter.js

```diff
@@ -0,0 +1,30 @@
+/* eslint-disable no-console */
+
+const fs = require('fs');
+const imagemin = require('imagemin');
+const imageminWebp = require('imagemin-webp');
+const path = require('path');
+
+
+// Recursively converts all jpg and png files to webp in docs directory tree
+(async () => {
+    async function convertImagesToWebp(dirPath) {
+        // Converts all jpg and png files in current directory to webp
+        await imagemin([`${dirPath}/*.{jpg,png}`], {
+            destination: dirPath,
+            plugins: [
+                imageminWebp({ quality: 65 }),
+            ],
+        });
+
+        const files = fs.readdirSync(dirPath);
+
+        // Find directories in current directory and convert their images to webp
+        await Promise.all(files.filter(file => fs.statSync(`${path.join(path.resolve(path.dirname('')), dirPath)}/${file}`).isDirectory())
+            .map(async directory => convertImagesToWebp(`${dirPath}/${directory}`)));
+    }
+    console.log('Webp conversion START');
+    // Start converting from docs folder
+    await convertImagesToWebp('docs');
+    console.log('Webp conversion DONE');
+})();
```

Comment:
You are running this main async function but there is no try/catch so it may not even fail the GitHub action in a case of an error. It will just log unhandled promise rejection but wont fail the execution with current Node version. To properly handle It you can do one of:

```
(async function () {
  try {

  } catch (err) {
    log.exception(err, ....);
    process.exit(1);
  }
})();
```

```
(async function () {

})().catch((err) => {
    log.exception(err, ....);
    process.exit(1);
});
```